### DISCUSSION TITLE
Where to set API key?

### DISCUSSION THREAD

**Discussion by ADevAJ** [darrenburns/elia#12] [Questions]
Just Used pipx on Windows 10 to install , was successful , but received error on
export OPENAI_API_KEY="xxx..." (I replaced with my key)

'export' is not recognized as an internal or external command,
operable program or batch file. 

When I run elia it opens but any message results in error : "No API key provided." 

I am not sure where to put my key. I looked for a json or text file and did not see an obviously named one. Please tell me where to insert! Not a wiz with pipx so I'd prefer a place to just copy/paste it into code.

Idea: perhaps add a textual modal screen within Elia where one can paste the api key?

Very excited to use this!

**Comment by juftin** [comment]
`export OPENAPI_API_KEY=xxxx` is setting an environment variable on Mac/Linux systems. This process works differently on Windows machines. Not being a Windows user myself, I don't exactly know how this works. Try the good old google machine if these steps from ChatGPT don't work:

1. Open the Start Menu: Click on the Windows Start button in the bottom-left corner of your screen.
2. Search for "Environment Variables": In the search bar, type "Environment Variables" and select "Edit the system environment variables" from the search results.
3. Click "Environment Variables": In the "System Properties" window that appears, click the "Environment Variables" button near the bottom right. 
4. Add a New User Environment Variable:
5. In the "User variables" section at the top, click the "New" button.
6. In the "Variable name" field, enter the name of your variable, in this case, "OPENAI_API_KEY."
7. In the "Variable value" field, enter the API key you want to set, replacing "xxx..." with your actual key.
8. Click OK: After entering the variable name and value, click the "OK" button to save the environment variable.
9. Close the Windows: Close the "Environment Variables" and "System Properties" windows.

**Comment by ADevAJ** [comment]
Thank you for the reply and tips! I will try the registry process and write back here with result

**Comment by Kabouik** [comment]
Would it make sense to allow using a configuration file in `~/.config/elia` as an alternate way to set the API key? My shell .rc file may be shared publicly in git repositories and I don't necessarily want to set the API key at run time every time I run `elia`.

Great TUI by the way! I always wanted that, even tried myself but failed miserably. I'm glad you did  such an awesome job!
